src/main/java/org/assertj/core/internal/Files.java

```diff
@@ -88,10 +89,15 @@ public static Files instance() {
   public void assertSameContentAs(AssertionInfo info, File actual, File expected) {
     verifyIsFile(expected);
     assertIsFile(info, actual);
+    BinaryDiffResult binaryDiffResult = null;
     try {
+      binaryDiffResult = binaryDiff.diff(actual, java.nio.file.Files.readAllBytes(expected.toPath()));
+      if (binaryDiffResult.hasNoDiff()) return;
       List<Delta<String>> diffs = diff.diff(actual, expected);
       if (diffs.isEmpty()) return;
       throw failures.failure(info, shouldHaveSameContent(actual, expected, diffs));
+    } catch (MalformedInputException e) {
+      throw failures.failure(info, shouldHaveBinaryContent(actual, binaryDiffResult));
```

Comment:
I did that because when you have a binary file which cannot be readLine(), you get an ugly exception : 
`Unable to compare contents of files:</tmp/junit6310584610408013618/junit1823075884829217683.tmp> and:</tmp/junit6310584610408013618/junit174924993450801969.tmp>` whereas the binary diff is able to say where the file differ, bit-wise (I think it's better to have an error saying that the files are binary differents than en error saying that we cannot compare the contents of the files). If you dont agree I can remove this part, just keep in mind that MalformedInpotException is only thrown by the `reader.readLine()` which is used only in the diff part (not the binaryDiff)